src/kubernetes_cluster/spec/pod_event/types.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::kubernetes_api_objects::spec::{dynamic::*, resource::*};
+use crate::kubernetes_cluster::spec::message::*;
+use crate::state_machine::action::*;
+use crate::state_machine::state_machine::*;
+use crate::temporal_logic::defs::*;
+use vstd::{multiset::*, prelude::*};
+
+verus! {
+
+pub struct PodEventState {
```

Comment:
I suggest we make `pod_event_enabled` a field in the cluster state of the top level state machine instead of here. And the PodEventState can be just a `()`.